txs/cache.go

```diff
@@ -553,15 +553,10 @@ func (c *Cache) cleanupAccounts(accounts ...types.Address) {
 	}
 }
 
-//   - errInsufficientBalance:
-//     conservative cache is conservative in that it only counts principal's spending for pending transactions.
-//     a tx rejected due to insufficient balance MAY become feasible after a layer is applied (principal
-//     received incoming funds). when we receive a errInsufficientBalance tx, we should store it in db and
-//     re-evaluate it after each layer is applied.
 //   - errTooManyNonce: when a principal has way too many nonces, we don't want to blow up the memory. they should
 //     be stored in db and retrieved after each earlier nonce is applied.
 func acceptable(err error) bool {
-	return err == nil || errors.Is(err, errInsufficientBalance) || errors.Is(err, errTooManyNonce)
+	return err == nil || errors.Is(err, errTooManyNonce)
 }
```

Comment:
Could you explain in the PR description from a users perspective what this change will accomplish? I assume before it was possible to add transactions to the mempool that would overspend an accounts balance, they just weren't added to a block until that account had sufficient balance, while now transactions that overspend will be ignored?